app/services/process_payment_intent.rb

```diff
@@ -20,7 +20,7 @@ def call!
 
     last_payment.update_attribute(:cvv_response_message, nil)
     OrderWorkflow.new(@order).next
-    last_payment.complete! if !last_payment.completed?
+    last_payment.complete! if %w('pending processing').include?(last_payment.state)
```

Comment:
I would personally make the effort to find an abstract name for this logic so that 1) we don't mix levels of abstraction 2) we make it easier for the next dev to understand why `pending` and `processing` states and not others.

Reply:
Good catch, I'm not sure I didn't just use the `can_complete?` method in the first place.